src/lib.rs

```diff
@@ -392,10 +420,11 @@ fn render_fn(ui: &Ui) {
     advance_logs(&mut logs);
     update_logs(&mut logs);
 
-    let show_window = Settings::get().show_window();
+    let mut settings = Settings::get_mut();
+    let mut show_window = settings.show_window();
     if show_window {
         Window::new(e("Log Uploader"))
-            .opened(&mut Settings::get_mut().show_window)
+            .opened(&mut show_window)
```

Comment:
since you already keep the lock around for the whole window, you can just use `&mut settings.show_window` here, same for the `if` on line 425. the write on line 453 should not be needed then

Reply:
When I tried that it I get a borrow error in the `.build()` due to the `&mut settings.copy_success` for the checkbox on line 439.